--- a/configure.ac
+++ b/configure.ac
@@ -729,7 +729,6 @@ case "${host}" in
 	fi
 	case "${host}" in
 	  *-*-cygwin*)
-	    DUMP_SYMS="dumpbin /SYMBOLS"
 	    ;;
 	  *)
 	    ;;
--- a/include/jemalloc/internal/jemalloc_internal_decls.h
+++ b/include/jemalloc/internal/jemalloc_internal_decls.h
@@ -19,7 +19,6 @@
 #  include <sys/param.h>
 #  include <sys/mman.h>
 #  if !defined(__pnacl__) && !defined(__native_client__)
-#    include <sys/syscall.h>
 #    if !defined(SYS_write) && defined(__NR_write)
 #      define SYS_write __NR_write
 #    endif
